ci-operator/step-registry/assisted/baremetal/operator/catalog-publish/assisted-baremetal-operator-catalog-publish-commands.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ baremetalds assisted operator catalog publish command ************"
+
+# Setup registry credentials
+echo "## Setting up registry credentials."
+mkdir -p "$HOME/.docker"
+REGISTRY_TOKEN_FILE="$SECRETS_PATH/$REGISTRY_SECRET/$REGISTRY_SECRET_FILE"
+cp "${REGISTRY_TOKEN_FILE}" "$HOME/.docker/config.json"
+
+set -x
+
+# Install tools
+echo "## Install opm"
+curl -L https://github.com/operator-framework/operator-registry/releases/download/v1.21.0/linux-amd64-opm -o /tmp/opm && chmod +x /tmp/opm
+echo "   opm installed"
+
+echo "## Install umoci"
+curl -L https://github.com/opencontainers/umoci/releases/download/v0.4.7/umoci.amd64 -o /tmp/umoci && chmod +x /tmp/umoci
+echo "   umoci installed"
+
+echo "## Install python dependencies"
+python3 -m pip install --user --upgrade pip
+python3 -m pip install --user -r deploy/olm-catalog/requirements.txt
+echo "   python dependencies installed"
+
+BUNDLE_DESTINATION="${REGISTRY_HOST}/${REGISTRY_ORG}/${REGISTRY_BUNDLE_REPOSITORY_NAME}:${REGISTRY_BUNDLE_REPOSITORY_TAG}"
+echo "## Generate and publish bundle to ${BUNDLE_DESTINATION}"
+
+SRC_DIR="${PWD}"
+
+tmp=$(mktemp -d)
+cd "${tmp}"
+
+/tmp/umoci init --layout bundle
+/tmp/umoci new --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.mediatype.v1=registry+v1' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.mediatype.v1=registry+v1' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.manifests.v1=manifests/' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.metadata.v1=metadata/' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.package.v1=assisted-service-operator' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.channels.v1=alpha,ocm-2.5' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.bundle.channel.default.v1=alpha' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.metrics.builder=operator-sdk-v1.10.0+git' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.metrics.mediatype.v1=metrics+v1' --image bundle:latest
+/tmp/umoci config --config.label 'operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v3' --image bundle:latest
+/tmp/umoci unpack --rootless --image bundle:latest bundle
```

Comment:
Ideally we should probably put this inside script files in the service repo, but for now let's merge this